lib/fixed-points.nix

```
@@ -53,42 +53,155 @@ rec {
       else converge f x';
 
   /*
-    Modify the contents of an explicitly recursive attribute set in a way that
-    honors `self`-references. This is accomplished with a function
+    Extend a function using an overlay.
+
+    Overlays allow modifying and extending fixed-point functions, specifically ones returning attribute sets.
+    A fixed-point function is a function which is intended to be evaluated by passing the result of itself as the argument.
+    This is possible due to Nix's lazy evaluation.
+
+
+    A fixed-point function returning an attribute set has the form
 
     ```nix
-    g = self: super: { foo = super.foo + " + "; }
+    final: { # attributes }
+    ```
+
+    where `final` refers to the lazily evaluated attribute set returned by the fixed-point function.
+
+    An overlay to such a fixed-point function has the form
+
+    ```nix
+    final: prev: { # attributes }
+    ```
+
+    where `prev` refers to the result of the original function to `final`, and `final` is the result of the composition of the overlay and the original function.
+
+    Applying an overlay is done with `extends`:
```

I think it'd be easier to read out of context of this file if we say `lib.extends` instead of `extends`:
```suggestion
    Applying an overlay is done with `lib.extends`:
```
(here and in the rest of the comment) 

The same way you're mentioning `lib.fix` below instead of just `fix` (although you're using `fix` not `lib.fix` in the code example..)

I'd suggest to also change the code examples to mention `lib.` for better readability, wdyt?